pkg/spoke/submarineragent/config_controller.go

```diff
@@ -368,15 +402,8 @@ func (c *submarinerConfigController) unlabelNode(ctx context.Context, config *co
 
 func (c *submarinerConfigController) addGateways(ctx context.Context, config *configv1alpha1.SubmarinerConfig,
 	expectedGateways int) ([]string, error) {
-	var zoneLabel string
-	// currently only gcp is supported
-	switch config.Status.ManagedClusterInfo.Platform {
-	case "GCP":
-		zoneLabel = gcpZoneLabel
-	default:
-		// for other non-public cloud platform (vsphere) or native k8s
-		zoneLabel = defaultZoneLabel
-	}
+	// for other non-public cloud platform (vsphere) or native k8s
+	zoneLabel := defaultZoneLabel
```

Comment:
I think we can implement similar functionality in cloud-prepare where we just tag nodes as Gatewaynodes and assume that the firewall is properly configured out of band. Once it is implemented, can we remove this code and use cloud-prepare?

Reply:
yes that can be done. This code is retained just for vmware. GCP/AWS does not use it now.